EugeneChai/hr analytics.Rmd

```diff
@@ -0,0 +1,272 @@
+check_in<-read.csv("Datasets/in_time.csv")
+check_out<-read.csv("Datasets/out_time.csv")
+```
+
+# Duplicates
+```{r}
+employee_survey_data$EmployeeID%>%unique()%>%length()
+general_data$EmployeeID%>%unique()%>%length()
+manager_survey_data$EmployeeID%>%unique()%>%length()
+check_in$X%>%unique()%>%length()
+check_out$X%>%unique()%>%length()
+#No duplicate rows
+
+```
+
+
+# Join all the datasets into one and create an overtime variable
+
+A potentially useful explanatory variable that may affect the attribution rate is whether an employee works overtime or not. This can be calculated by manipulating the check-in and check-out data.
+```{r}
+check_in_na<-as.tibble(t(colSums(is.na(check_in))))
+#If all the employees did not check-in on a particular day, it can be assumed that the office is closed on that day. 
+closed_dates<-check_in_na%>%pivot_longer(cols=colnames(check_in_na),names_to="Dates",values_to="Number_of_null_values")%>%filter(Number_of_null_values==4410)
+closed_dates<-unlist(closed_dates[,1])
+```
+
+```{r}
+#Remove all dates when the office is closed
+check_in<-check_in[,!(names(check_in)%in% closed_dates)]
+check_out<-check_out[,!(names(check_out)%in%closed_dates)]
+```
+
+```{r}
+all_dates<-names(check_in)[2:length(names(check_in))]
+```
+
+```{r}
+check_in_long<-check_in%>%pivot_longer(cols=all_dates,names_to="Dates",values_to="check_in_time")%>%mutate(id=str_c(X,Dates,sep=""))%>%select(X,id,check_in_time)
+check_out_long<-check_out%>%pivot_longer(cols=all_dates,names_to="Dates",values_to="check_out_time")%>%mutate(id=str_c(X,Dates,sep=""))%>%select(id,check_out_time)
+
+```
+
+```{r}
+#Ensure that the check-in and check-out time variables are on the same dataframe and are converted into date objects. 
+check_in_out<-check_in_long%>%inner_join(check_out_long,by="id")%>%mutate(check_in_time=as.POSIXlt(check_in_time),check_out_time=as.POSIXlt(check_out_time))
+```
+
+```{r}
+#Create a new column that represents the average working hours for each employee.
+check_in_out<-check_in_out%>%mutate(working_hours=difftime(check_out_time,check_in_time,units="hours"))
+```
+
+```{r}
+check_in_out<-check_in_out%>%group_by(X)%>%summarise(avg_working_hours=mean(working_hours,na.rm=TRUE))%>%rename(EmployeeID=X)
+```
+
+Join all the relevant data together into one dataframe
+```{r}
+df<-general_data%>%inner_join(manager_survey_data,by="EmployeeID")%>%inner_join(employee_survey_data,by="EmployeeID")%>%inner_join(check_in_out,by="EmployeeID")
+```
+
+```{r}
+#Add a new column that determines if an employee has worked overtime or not
+df<-df%>%mutate(overtime=if_else(avg_working_hours>8,"Yes","No"))
+```
+
+
+# Data Imputation
+```{r}
+colSums(is.na(df))
+```
+```{r}
+#The null values in NumCompaniesWorked can be filled by using estimated values formed by using both YearsAtCompany and TotalWorkingYears.
+na_numcompaniesworked<-df %>% filter(is.na(NumCompaniesWorked))%>%select(EmployeeID,NumCompaniesWorked,YearsAtCompany,TotalWorkingYears)

```

Comment:
Does this use YearsAtCompany and TotalWorkingYears to impute the nulls? This seems like a very interesting approach, I'm curious as to your thinking behind this :)